src/Microsoft.DocAsCode.Build.Engine/Incrementals/IncrementalBuildContext.cs

```diff
@@ -557,6 +554,43 @@ private bool GetCanVersionIncremental(IncrementalStatus buildInfoIncrementalStat
             return true;
         }
 
+        private IEnumerable<FileItem> GetFilesToCalculateAttributes()
+        {
+            var keys = new HashSet<string>();
+            foreach (var f in _parameters.Files.EnumerateFiles())
+            {
+                var fileKey = ((RelativePath)f.File).GetPathFromWorkingFolder().ToString();
+                keys.Add(fileKey);
+                yield return new FileItem
+                {
+                    PathFromWorkingFolder = fileKey,
+                    FullPath = f.FullPath,
+                    IsFromSource = true,
+                };
+            }
+            if (CurrentBuildVersionInfo.Dependency != null)
+            {
+                foreach (var f in CurrentBuildVersionInfo.Dependency.GetAllDependentNodes())
+                {
+                    var p = RelativePath.TryParse(f);
+                    if (p == null)
+                    {
+                        continue;
+                    }
+                    var fullPath = PathUtility.GetFullPath(EnvironmentContext.BaseDirectory, p.RemoveWorkingFolder());
+                    if (!keys.Contains(f) && File.Exists(fullPath))
```

Comment:
yes. if the user writes a link that doesn't exist at all, we don't need to calculate the md5 info for it.

---
In reply to: [93815073](https://github.com/dotnet/docfx/pull/1048#discussion_r93815073) [](ancestors = 93815073)